### ISSUE TITLE
Lacking documentation for (or bugs in?) specifying alpha values in ColorMapper's palette

### ISSUE THREAD

**Issue by datacticaempii** [bokeh/bokeh#8320]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
$ bokeh --version
0.13.0
$ python --version
Python 3.7.0
$ uname -a
Linux greenbean 4.18.9-arch1-1-ARCH #1 SMP PREEMPT Wed Sep 19 21:19:17 UTC 2018 x86_64 GNU/Linux
Firefox: 62.0.3 (64-bit)

#### Description of expected behavior and the observed behavior

I've had troubles defining alpha channel for Image's color mapper. The [doc](https://bokeh.pydata.org/en/latest/docs/reference/models/glyphs/image.html) states that "A ColorMapper to use to map the scalar data from image into RGBA values for display.". Similarly, the documentation for parameter global_alpha suggests that alpha can be defined in the color mapper.

Expected: documentation (for ColorMapper?) should show an example for specifying the alpha. Or the documented "RGB(A) hex value" or "CSS-format RGB/RGBA strings" should be accepted without ValueErrors.

[Another doc](https://bokeh.pydata.org/en/latest/docs/user_guide/styling.html#specifying-colors) lists different ways to specify colors, but this seems to be outdated (or there are bugs in the implementation?):
 * The doc states that "an RGB(A) hex value, e.g., '#FF0000', '#44444444'" is accepted. But apparently this syntax is not supported when defining a palette.
 * The warning in the doc that refers to issue 2622 claims that "CSS-format RGB/RGBA strings" can be used. But I got ValueError when trying ("case 4" in the example code below):
    expected an element of Seq(Color), got seq with invalid items ['#00000000', '#00ff00cc']
    expected an element of Seq(Color), got seq with invalid items ['rgb(0,0,0)', 'rgb(0,255,0)']
    expected an element of Seq(Color), got seq with invalid items ['rgb(0, 0, 0, 0.1)', 'rgb(0, 255, 0, 0.8)']
    expected an element of Seq(Color), got seq with invalid items ['rgba(0, 0, 0, 0.1)', 'rgba(0, 255, 0, 0.8)']

When trying different ways, I actually found a working approach: specifying the colors like `bokeh.colors.RGB(0,0,0, 0.1)` seems to work. The only minor issue is a bogus error message when running `bokeh serve`:

    2018-10-12 12:12:39,898 error handling message Message 'PATCH-DOC' (revision 1): ValueError("expected an element of Seq(Color), got seq with invalid items ['rgb(0, 0, 1)', 'rgb(0, 255, 1)']")

This message seems to appear in the log only for the first palette that uses RGB objects: If I comment out the offending line in the example code ("Case 1"), similar message appears for "Case 2". But the colors get displayed correctly.

#### Example code that reproduces the issue

This code can be run as `python bug.py` or `bokeh serve --show bug.py`.

```
import numpy as np

from bokeh.colors import RGB
from bokeh.io import curdoc, show
from bokeh.plotting import figure
from bokeh.models.mappers import LinearColorMapper

buf = np.triu(255*np.ones((200,200), dtype=np.uint8))
p = figure(background_fill_color='red')

def img(cm, x, y):
    p.image(image=[buf], color_mapper=LinearColorMapper(palette=cm, low=0, high=255),
            x=[x], y=[y], dh=[1], dw=[1])

# Case 1. These are OK: opaque black/green rectangle
img(['#000000', '#00ff00'], 0, 0)
img([RGB(0,0,1), RGB(0,255,1)], 0, 1)

# Case 2. This works! Getting the expected transparency
img([RGB(0,0,2,0.1), RGB(0,255,2,0.8)], 1, 0)

# Case 3. Bug https://github.com/bokeh/bokeh/issues/2622 (?): no image shown
img([(0,0,3), (0,255,3)], 2, 0)
img([(0, 0, 4, 0.1), (0, 255, 4, 0.8)], 2, 1)

# Case 4. These give ValueError: expected an element of Seq(Color), got seq with invalid items [...]
for y,cm in enumerate([['#00000000', '#00ff00cc'],
                       ['rgb(0,0,0)', 'rgb(0,255,0)'],
                       ['rgb(0, 0, 0, 0.1)', 'rgb(0, 255, 0, 0.8)'],
                       ['rgba(0, 0, 0, 0.1)', 'rgba(0, 255, 0, 0.8)'],
                       ]):
    try:
        img(cm, 3, y)
    except ValueError as ex:
        print(ex)

if __name__ == '__main__':
    show(p)
else:
    # Using 'bokeh serve'
    curdoc().add_root(p)
```

**Comment by bryevdv** [comment]
The script no longer generates any errors on running after #8371, but I am not sure what the expected output is supposed to look like. Here is the output of just case 4:

<img width="608" alt="screen shot 2018-10-28 at 18 46 28" src="https://user-images.githubusercontent.com/1078448/47625374-fbb15600-dae1-11e8-9572-0d5f4186867f.png">

**Comment by philippjfr** [comment]
Going through them manually I get the following behavior

>['#00000000', '#00ff00cc']

Not supported, could add some logic to ``Color.transform`` to turn these into ``rgba`` strings.

>['rgb(0,0,0)', 'rgb(0,255,0)']

Works fine.

>['rgb(0, 0, 0, 0.1)', 'rgb(0, 255, 0, 0.8)']

Not valid since the new regex I added does not accept alpha values in an ``rgb`` string. Could lift that restriction since it seems browsers will accept the alpha value in this case.

>['rgba(0, 0, 0, 0.1)', 'rgba(0, 255, 0, 0.8)']

Works fine now that my PR has been merged.
